niworkflows/utils/misc.py

```diff
@@ -184,6 +184,19 @@ def splitext(fname):
     return stem, basename[len(stem):]
 
 
+def select_first(in_files):
+    """
+    Select the first file from a list of filenames.
+    Used to grab the first echo's file when processing
+    multi-echo data through workflows that only accept
+    a single file.
+    """
+    if isinstance(in_files, list):
+        return in_files[0]
+    else:
+        return in_files
```

Comment:
I'm a bit surprised we don't already have this somewhere. While we're here, could you just add a doctest showing the results of both conditions?